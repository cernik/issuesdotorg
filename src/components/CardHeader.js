// @flow
import React from 'react';
import {Text, View, StyleSheet} from 'react-native';

import OpenWhite from '../assets/svg/open-white.svg';
import ClosedWhite from '../assets/svg/closed-white.svg';

import {Styles} from '../utils/styles';
import type {ItemType} from '../types';

type CardHeaderProps = {
  item: ItemType,
};

const CardHeader = ({item}: CardHeaderProps) => {
  const isClosed = item.state === 'closed';
  return (
    <>
      <Text style={Styles.titleBig}>
        {item.title}
        <Text
          style={{
            fontWeight: '500',
            color: 'grey',
          }}>{` #${item?.number}`}</Text>
      </Text>
      <View
        style={{
          paddingVertical: 16,
          marginBottom: 16,
          alignItems: 'center',
          flexDirection: 'row',
          borderBottomWidth: 1,
          borderColor: 'gainsboro',
        }}>
        <View
          style={[
            Styles.row,
            styles.state,
            {
              backgroundColor: isClosed ? 'red' : 'green',
            },
          ]}>
          {isClosed ? (
            <ClosedWhite style={Styles.icon} />
          ) : (
            <OpenWhite style={Styles.icon} />
          )}
          <Text
            style={{
              fontSize: 18,
              color: 'white',
              fontWeight: '600',
              marginHorizontal: 8,
            }}>{`${isClosed ? 'Closed' : 'Open'}`}</Text>
        </View>
        <Text style={{fontSize: 16, marginStart: 8}}>{`${
          item.comments
        } comment${item.comments > 1 ? 's' : ''}`}</Text>
      </View>
    </>
  );
};

const styles = StyleSheet.create({
  state: {
    paddingHorizontal: 12,
    height: 32,
    borderRadius: 16,
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default CardHeader;
