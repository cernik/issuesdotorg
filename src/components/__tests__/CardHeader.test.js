// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import CardHeader from '../CardHeader';

describe('snapshot', () => {
  const item = {
    id: 808245666,
    state: 'closed',
    title: 'title goes here',
    body: 'default body',
    number: 704,
    comments: 12,
    user: {
      login: 'johndoe110',
      avatar_url: 'path_to_avatar.png',
    },
    created_at: '2021-02-15T07:02:04Z',
    updated_at: '2021-02-15T07:12:37Z',
  };

  it('render correctly with given data', () => {
    const tree = renderer.create(<CardHeader item={item} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
