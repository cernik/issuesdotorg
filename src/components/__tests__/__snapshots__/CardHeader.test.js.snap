// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot render correctly with given data 1`] = `
Array [
  <Text
    style={
      Object {
        "fontSize": 22,
        "fontWeight": "700",
      }
    }
  >
    title goes here
    <Text
      style={
        Object {
          "color": "grey",
          "fontWeight": "500",
        }
      }
    >
       #704
    </Text>
  </Text>,
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "gainsboro",
        "flexDirection": "row",
        "marginBottom": 16,
        "paddingVertical": 16,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          Object {
            "alignItems": "center",
            "borderRadius": 16,
            "height": 32,
            "justifyContent": "center",
            "paddingHorizontal": 12,
          },
          Object {
            "backgroundColor": "red",
          },
        ]
      }
    >
      <SvgMock
        style={
          Object {
            "margin": 2,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 18,
            "fontWeight": "600",
            "marginHorizontal": 8,
          }
        }
      >
        Closed
      </Text>
    </View>
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginStart": 8,
        }
      }
    >
      12 comments
    </Text>
  </View>,
]
`;
